pkg/ring/lifecycler_test.go

```diff
@@ -246,17 +256,90 @@ func TestCheckReady(t *testing.T) {
 	flagext.DefaultValues(&ringConfig)
 	ringConfig.KVStore.Mock = &MockClient{}
 
+	tokens := Tokens([]uint32{1})
 	r, err := New(ringConfig, "ingester")
 	require.NoError(t, err)
 	defer r.Stop()
-	cfg := testLifecyclerConfig(ringConfig, "ring1")
+	cfg := testLifecyclerConfig(t, ringConfig, "ring1")
 	cfg.MinReadyDuration = 1 * time.Nanosecond
 	l1, err := NewLifecycler(cfg, &nopFlushTransferer{}, "ingester")
-	l1.setTokens([]uint32{1})
+	l1.setTokens(tokens)
 	l1.Start()
 	require.NoError(t, err)
 
 	// Delete the ring key before checking ready
 	err = l1.CheckReady(context.Background())
 	require.Error(t, err)
 }
+
+type noopFlushTransferer struct {
+	lifecycler *Lifecycler
+}
+
+func (f *noopFlushTransferer) StopIncomingRequests()                 {}
+func (f *noopFlushTransferer) Flush()                                {}
+func (f *noopFlushTransferer) TransferOut(ctx context.Context) error { return nil }
+
+func TestTokensOnDisk(t *testing.T) {
+	var ringConfig Config
+	flagext.DefaultValues(&ringConfig)
+	ringConfig.KVStore.Mock = consul.NewInMemoryClient(GetCodec())
+
+	r, err := New(ringConfig, "ingester")
+	require.NoError(t, err)
+	defer r.Stop()
+
+	lifecyclerConfig := testLifecyclerConfig(t, ringConfig, "ing1")
+	lifecyclerConfig.NumTokens = 512
+
+	// Start first ingester.
+	ft := &noopFlushTransferer{}
+	l1, err := NewLifecycler(lifecyclerConfig, ft, "ingester")
```

Comment:
You can directly pass `&noopFlushTransferer{}` instead of `ft`, like do for the ingester 2.